backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramEventProcessor.java

```diff
@@ -144,8 +143,7 @@ public void refresh(ChangeDescription changeDescription) {
      * @return <code>true</code> if the diagram should be refreshed, <code>false</code> otherwise
      */
     private boolean shouldRefresh(ChangeDescription changeDescription) {
-        return ChangeKind.SEMANTIC_CHANGE.equals(changeDescription.getKind())
-                || (DiagramChangeKind.DIAGRAM_LAYOUT_CHANGE.equals(changeDescription.getKind()) && changeDescription.getSourceId().equals(this.diagramContext.getDiagram().getId()));
+        return this.diagramCreationService.shouldRefresh(this.editingContext, this.diagramContext, changeDescription);
```

Comment:
Do not rely on the diagramCreationService for that

Reply:
Done